### PR TITLE
Replace Travis CI with GitHub Actions

### LINKED ISSUE #585: Move from Travis to GitHub Action
## Problem description

As Travis is getting more and more obsolete, lets try GitHub Action.

## Expected behavior

A GitHub Action which does basically the same as the Travis CI.

### PR DESCRIPTION
**Pull request by tomschr**
This PR fixes #585 and contains:

* Remove `.travis.yml` and `.travis-debug`
* Create GitHub Action workflow
* Content of `.travis-debug` is integrated in `ci.yml` now
* Use branches-ignore on `gh-pages`

### CODE REVIEW

**Comment by tomschr** [comment]
I've investigated this PR again:

1. All the guides under `doc/` are correctly validated and built.
2. The manpage under `man/` doesn't build. The `mdash` entity is unknown.

I suspect a catalog issue but couldn't find the culprit. The `mdash`entity is defined in `http://www.w3.org/2003/entities/iso8879/isopub.ent`. So to solve this stupid validation step, I've included that entity into `daps-init.xml` and voila! It works.

I'm still not sure where the cause is, but I leave it for the time being. This PR is not about entities, it's about the GitHub Action.

@fsundermeyer + @sknorr I think it's in a state to merge it into main. Let me know if this can be improved.

File: man/xml/daps-init.xml [hunk 1]
```diff
@@ -6,6 +6,11 @@
   <!ENTITY daps_init "daps-init">
   <!ENTITY % entities SYSTEM "entity-decl.ent">
   %entities;
+  <!-- Needed for GitHub Action, see openSUSE/daps#585 -->
```
Comment (ghost) [inline man/xml/daps-init.xml]: #585 does not explain the added entities below this comment... this comment seems misleading

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                              python-libxml2 \
+                              python-lxml \
+                              python3 \
+                              python3-lxml \
+                              remake \
+                              sgml-base \
+                              sgml-data \
+                              shtool \
+                              shunit2 \
+                              transfig \
```
Comment (ghost) [inline .github/workflows/ci.yml]: We probably don't need that anymore:

```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                              libtool \
+                              libxml-commons-external-java \
+                              libxml2 \
+                              libxml2-utils \
+                              libxmlgraphics-commons-java \
+                              libxslt1.1 \
+                              optipng \
+                              poppler-utils \
+                              python-libxml2 \
+                              python-lxml \
```
Comment (ghost) [inline .github/workflows/ci.yml]: We shouldn't need these anymore:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      - name: Configure
+        run: ./configure --sysconfdir=/etc
+
+      - name: make
+        run: make
+
+      - name: make install
+        run: sudo make install
+
+      - name: Running daps XML syntax check for double IDs...
```
Comment (ghost) [inline .github/workflows/ci.yml]: Language improvement:

```suggestion
      - name: Running daps XML syntax check for duplicated IDs...
```

**Review by ghost — COMMENTED** [review]
Looks good, thanks! Small changes below ...

File: man/xml/daps-init.xml (hunk 1 above)
Comment (tomschr) [reply man/xml/daps-init.xml]: Yeah. Maybe it's a better idea to use the pull request number instead of the issue. I've corrected that in commit 7183b653.

**Comment by tomschr** [comment]
Thanks! Much appreciated. :+1:

**Pull request merged**